firebase-sessions/src/main/kotlin/com/google/firebase/sessions/api/SessionSubscriber.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions.api
+
+/** [SessionSubscriber] is an interface that dependent SDKs must implement. */
+interface SessionSubscriber {
+  /** [SessionSubscriber.Name]s are used for identifying subscribers. */
+  enum class Name {
+    CRASHLYTICS,
+    PERFORMANCE,
+  }
+
+  /** [SessionDetails] contains session data passed to subscribers whenever the session changes */
+  data class SessionDetails(val sessionId: String)
+
+  fun onSessionChanged(sessionDetails: SessionDetails)
```

Comment:
This is a different SessionDetails that currently only contains the sessionId. This is the same as what iOS does. We can change it to just a string of a sessionId, but I like the idea of it being a container like this because we can add more fields in the future without breaking the api.

Reply:
Removed exception anyway.